lang/cs/Org.Apache.REEF.Client/AzureBatch/Util/JobJarMaker.cs

```diff
@@ -43,7 +48,14 @@ internal sealed class JobJarMaker
         /// <returns>A string path to file.</returns>
         public string CreateJobSubmissionJAR(JobRequest jobRequest)
         {
-            throw new NotImplementedException();
+            string localDriverFolderPath = CreateDriverFolder(jobRequest.JobIdentifier);
+            _driverFolderPreparationHelper.PrepareDriverFolder(jobRequest.AppParameters, localDriverFolderPath);
+            return _resourceArchiveFileGenerator.CreateArchiveToUpload(localDriverFolderPath);
+        }
+
+        private string CreateDriverFolder(string jobId)
+        {
+            return Path.GetFullPath(Path.Combine(Path.GetTempPath(), string.Join("-", "reef", jobId)));
```

Comment:
Is job id unique with every invocation? If not, we need to make sure that the directories are unique between each invocation.

Reply:
yes. it is unique. Defined by "HelloworldJob-{datetime}"